src/Authorization.AspNetCore/AuthorizationValidationRule.cs

```diff
@@ -86,15 +86,15 @@ private async Task AuthorizeAsync(INode node, IProvideMetadata type, ValidationC
             if (policyNames?.Count == 1)
             {
                 // small optimization for the single policy - no 'new List<>()', no 'await Task.WhenAll()'
-                var authorizationResult = await _authorizationService.AuthorizeAsync(_httpContextAccessor.HttpContext.User, policyNames[0]);
+                var authorizationResult = await _authorizationService.AuthorizeAsync(_claimsPrincipalAccessor.GetClaimsPrincipal(context), policyNames[0]);
                 AddValidationError(node, context, operationType, authorizationResult);
             }
             else if (policyNames?.Count > 1)
             {
                 var tasks = new List<Task<AuthorizationResult>>(policyNames.Count);
                 foreach (string policyName in policyNames)
                 {
-                    var task = _authorizationService.AuthorizeAsync(_httpContextAccessor.HttpContext.User, policyName);
+                    var task = _authorizationService.AuthorizeAsync(_claimsPrincipalAccessor.GetClaimsPrincipal(context), policyName);
```

Comment:
Please, move `_claimsPrincipalAccessor.GetClaimsPrincipal(context)` out of the loop.